backend/pkg/api/handle_topic_create_integration_test.go

```diff
@@ -0,0 +1,735 @@
+// Copyright 2022 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file https://github.com/redpanda-data/redpanda/blob/dev/licenses/bsl.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+//go:build integration
+
+package api
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"math"
+	"math/rand"
+	"net/http"
+	"runtime"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/cloudhut/common/logging"
+	"github.com/cloudhut/common/rest"
+	"github.com/go-chi/chi/v5"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/twmb/franz-go/pkg/kadm"
+	"github.com/twmb/franz-go/pkg/kerr"
+	"github.com/twmb/franz-go/pkg/kfake"
+	"github.com/twmb/franz-go/pkg/kgo"
+	"github.com/twmb/franz-go/pkg/kmsg"
+	"go.uber.org/zap"
+
+	"github.com/redpanda-data/console/backend/pkg/config"
+	"github.com/redpanda-data/console/backend/pkg/connect"
+	"github.com/redpanda-data/console/backend/pkg/console"
+	"github.com/redpanda-data/console/backend/pkg/kafka"
+	"github.com/redpanda-data/console/backend/pkg/redpanda"
+	"github.com/redpanda-data/console/backend/pkg/testutil"
+)
+
+type restAPIError struct {
+	Status  int    `json:"statusCode"`
+	Message string `json:"message"`
+}
+
+func (s *APIIntegrationTestSuite) TestHandleCreateTopic() {
+	t := s.T()
+	require := require.New(t)
+	assert := assert.New(t)
+
+	logCfg := zap.NewDevelopmentConfig()
+	logCfg.Level = zap.NewAtomicLevelAt(zap.InfoLevel)
+	log, err := logCfg.Build()
+	require.NoError(err)
+
+	port := rand.Intn(50000) + 10000
+
+	cfg := &config.Config{
+		REST: config.Server{
+			Config: rest.Config{
+				HTTPListenAddress: "0.0.0.0",
+				HTTPListenPort:    port,
+			},
+		},
+		Kafka: config.Kafka{
+			Brokers: []string{s.testSeedBroker},
+		},
+		Connect: config.Connect{
+			Enabled: false,
+		},
+		Logger: logging.Config{
+			LogLevelInput: "info",
+			LogLevel:      zap.NewAtomicLevel(),
+		},
+	}
+
+	api := New(cfg)
+
+	go api.Start()
+
+	kafkaClient, err := kgo.NewClient(kgo.SeedBrokers(s.testSeedBroker))
+	require.NoError(err)
+
+	kafkaAdminClient := kadm.NewClient(kafkaClient)
+
+	// allow for server to start
+	timer1 := time.NewTimer(10 * time.Millisecond)
+	<-timer1.C
+
+	apiServer := api.Cfg.REST.HTTPListenAddress + ":" + strconv.FormatInt(int64(port), 10)
```

Comment:
Shouldn't all of that go into the testsuite setup and then become a member of your `APIIntegrationTestSuite` struct so that you don't have to do this again for all the other API endpoints?